students/Gentner/2_2GentnerPWMsawtooth/2_2GentnerPWMsawtooth.ino

```diff
@@ -0,0 +1,57 @@
+// Turn on an LED using PWM
+// Turn on another LED using our own
+// PWM function
+
+const int yLED = 9;
+const int bLED = 10; // constant for my blue LED
+
+/* function: pwm
+ *  
+ * Practice writing functions and understanding 
+ * PWM by explicitly setting output high/low
+ * according to input parameters.
+ * Sets pin to HIGH, delays, sets to LOW, delays.
+ * 
+ * pin: Pin number to execute PWM on
+ * brightness: How bright the LED should be.
+ *             0 is off, 255 is maximum.
+ * period: period of pulse in microseconds
+*/
+void pwm(int pin, double brightness, double period){
+  // Calculate Ton and Toff
+  double Ton;  // using double precision to avoid overflow
+  double Toff;
+  Ton = (brightness * period) / 255;
+  Toff = period - Ton;
+  
+  // Turn on
+  digitalWrite(pin, HIGH);
+  // delay for Ton
+  delayMicroseconds(Ton);
+  // Turn off
+  digitalWrite(pin, LOW);
+  // delay for Toff
+  delayMicroseconds(Toff);
+}
+
+void setup() {
+  // set pin mode to output
+  pinMode(yLED, OUTPUT);
+  // set blue LED pin to output
+  pinMode(bLED, OUTPUT);
+}
+
+void loop() {
+ //double brightness = 7.84;
+ // double period = 2040;
+  for (int i = 0; i <= 255; i++) 
+  {
+    analogWrite(yLED, i);    // Generate the sawtooth wave by varying the PWM duty cycle
+    analogWrite(bLED,i);
+    delayMicroseconds(7843);
```

Comment:
Try to avoid hard-coded numbers. Even if this is found empirically, include a comment explaining what it is and why it is this value.